### PR TITLE
Update registered.py to skip registerations

### LINKED ISSUE #1830: [GSoC] Avoid registering tests datasets
Currently `tfds.list_builder` list [`dummy_mnist`](https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/testing/test_utils.py#L393) and other test-only datasets. We should make sure to avoid registering those datasets.

* Add `with tfds.core.registered.skip_registeration():` context manager which when applied skip dataset registration. Changes should go in `registered.py`. Add tests in `registered_test.py`.
* Apply this to the public API: https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/public_api.py#L24

```
with core.registered.skip_registeration():
  from tensorflow_datasets import testing
```

* Add a `test_dataset_registered.py` which check that `tfds.list_builder` do not contains `dummy_mnist` or other test datasets.

### PR DESCRIPTION
**Pull request by Eshan-Agarwal**
fixes #1830

### CODE REVIEW

**Comment by Eshan-Agarwal** [comment]
@Conchylicultor  please review

**Comment by vijayphoenix** [comment]
@Eshan-Agarwal See #1832 :sweat_smile:

**Comment by Eshan-Agarwal** [comment]
@vijayphoenix  😅 what should I do now ? I thought I worked on it !

**Comment by vijayphoenix** [comment]
We both think alike https://github.com/tensorflow/datasets/pull/1833/commits/4603c70f6e971cbdaeba0371a5702903d774133e :-)

**Comment by Eshan-Agarwal** [comment]
@Conchylicultor  Please review Update all tests and scripts as per instruction given.

File: tensorflow_datasets/core/registered.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 
 from absl import flags
 from absl import logging
+from contextlib import contextmanager
```
Comment (Conchylicultor) [inline tensorflow_datasets/core/registered.py:30]: Please look at other PR for review.

File: tensorflow_datasets/core/registered.py [hunk 2]
```diff
@@ -106,6 +107,14 @@
     version=r"[0-9]+\.[0-9]+\.[0-9]+",
 ))
 
+_SKIP_REGISTRATION = False
+
+@contextmanager
+def skip_registration():
+  global _SKIP_REGISTRATION
```
Comment (Conchylicultor) [inline tensorflow_datasets/core/registered.py:114]: CAPITALIZE means constant. Here you modify the variable so should be lower_case

Comment (Eshan-Agarwal) [reply tensorflow_datasets/core/registered.py:114]: Okay

File: tensorflow_datasets/core/registered.py (hunk 1 above)
Comment (Eshan-Agarwal) [reply tensorflow_datasets/core/registered.py:30]: Should I wrap it in lazy_import?

**Comment by Eshan-Agarwal** [comment]
@Conchylicultor  Please review this PR again

**Pull request closed without merging**